### ISSUE TITLE
Can I train AI If AI model is located in the another model’s forward? 

### ISSUE THREAD

**Issue by SungmanHong** [pytorch/pytorch#60349]
Hi, I have TWO model.
The model2 is located in the model1's forward, as follows.

```
model2 = torchvision.models.video.r3d_18(pretrained=False).to('cuda')

class model1(torch.nn.Module):
    def __init__(self):
        super(model1, self).__init__()
        self.ReLU = ReLU(400, 128, 1)
        self.QoS = nn.Linear(128, 1)

    def forward(self, pixel):
        pixel = model2(pixel.cuda())
        pixel = self.ReLU(pixel)
        pixel = self.QoS(pixel)
        return pixel
```

Now, I’m training model1 by loss.backward()
However, the model2, located in the model1’s forward, is trained automatically?
If not, How can I train the model 2?

My model 1's train code is located at below as follows.

```
for idx, (raw_data, len_frame, true_mos) in enumerate(train_module):
        raw_data = raw_data.to(device).float()
        true_mos = true_mos.to(device).float()
        optimizer.zero_grad()
        outputs = model1(raw_data)
        loss = criterion(outputs, true_mos)
        loss.backward()
        optimizer.step()
```
Thanks.

**Comment by simaki** [comment]
You may want to add model2 to an attribute of model1 in \_\_init\_\_. Then the [parameters](https://pytorch.org/docs/stable/generated/torch.nn.Module.html?highlight=parameters#torch.nn.Module.parameters) of model1 include that of model2 and so model2 would be trained.

fyi questions are supposed to be posted to https://discuss.pytorch.org/ as it's more likely you will get help there.
